PSPrintSDK/OLProductTemplate.m

```diff
@@ -170,66 +119,51 @@ + (NSArray *)templates {
-        case kOLTemplateTypeMiniPolaroids: return @[[NSURL URLWithString:@"https://s3.amazonaws.com/sdk-static/mini%20polaroids1%402x.jpg"], [NSURL URLWithString:@"https://s3.amazonaws.com/sdk-static/mini%20polaroids2%402x.jpg"]];
-        case kOLTemplateTypePolaroids: return @[[NSURL URLWithString:@"https://s3.amazonaws.com/sdk-static/polaroids2%402x.jpg"], [NSURL URLWithString:@"https://s3.amazonaws.com/sdk-static/polaroids3%402x.jpg"]];
-        case kOLTemplateTypeFrame2x2: return @[[NSURL URLWithString:@"https://s3.amazonaws.com/sdk-static/frames1%402x.jpg"]];
-        case kOLTemplateTypeFrame3x3: return @[[NSURL URLWithString:@"https://s3.amazonaws.com/sdk-static/frames1%402x.jpg"]];
-        case kOLTemplateTypeFrame4x4: return @[[NSURL URLWithString:@"https://s3.amazonaws.com/sdk-static/frames1%402x.jpg"]];
-        case kOLTemplateTypeFrame: return @[[NSURL URLWithString:@"https://s3.amazonaws.com/sdk-static/frames1%402x.jpg"], [NSURL URLWithString:@"https://s3.amazonaws.com/sdk-static/frames2%402x.jpg"]];
-        case kOLTemplateTypePostcard: return @[[NSURL URLWithString:@"https://s3.amazonaws.com/sdk-static/postcards1%402x.jpg"], [NSURL URLWithString:@"https://s3.amazonaws.com/sdk-static/postcards2%402x.jpg"]];
-        case kOLTemplateTypeLargeFormatA1: return @[[NSURL URLWithString:@"https://s3.amazonaws.com/sdk-static/A1.jpg"]];
-        case kOLTemplateTypeLargeFormatA2: return @[[NSURL URLWithString:@"https://s3.amazonaws.com/sdk-static/A2.jpg"]];
-        case kOLTemplateTypeLargeFormatA3: return @[[NSURL URLWithString:@"https://s3.amazonaws.com/sdk-static/A3.jpg"]];
-        case kOLTemplateTypeStickersSquare: return @[[NSURL URLWithString:@"https://s3.amazonaws.com/sdk-static/product_square-stickers-1%402x.jpg"], [NSURL URLWithString:@"https://s3.amazonaws.com/sdk-static/product_square-stickers-2%402x.jpg"]];
-        case kOLTemplateTypeStickersCircle: return @[[NSURL URLWithString:@"https://s3.amazonaws.com/sdk-static/product_rounded-stickers-1%402x.jpg"], [NSURL URLWithString:@"https://s3.amazonaws.com/sdk-static/product_rounded-stickers-2%402x.jpg"]];
-        default: return nil;
+- (NSString *)description {
+    NSMutableString *supportedCurrencies = [[NSMutableString alloc] init];
+    for (NSString *currency in self.costsByCurrencyCode) {
+        [supportedCurrencies appendFormat:@" %@", currency];
     }
-    return @[[NSURL URLWithString:@"https://s3.amazonaws.com/sdk-static/polaroids3%402x.jpg"], [NSURL URLWithString:@"https://s3.amazonaws.com/sdk-static/polaroids3%402x.jpg"]];
+    return [NSString stringWithFormat:@"%@%@ (%@)%@ quantity: %lu",self.enabled ? @"enabled " : @"disabled ", self.identifier, self.name, supportedCurrencies, (unsigned long) self.quantityPerSheet];
 }
 
-
 #pragma mark - NSCoding protocol methods
-
 - (void)encodeWithCoder:(NSCoder *)aCoder {
     [aCoder encodeObject:self.identifier forKey:kKeyIdentifier];
```

Comment:
NSCoding has always given me an uneasy feeling as I do feel I was abusing it early on, perhaps we should bite the bullet and move to SQLite database sooner rather than later?